src/entities/addLiquidity/addLiquidityV2/index.ts

```diff
@@ -50,4 +51,8 @@ export class AddLiquidityV2 implements AddLiquidityBase {
     ): AddLiquidityBuildCallOutput {
         return this.getAddLiquidity(input.poolType).buildCall(input);
     }
+
+    public buildCallWithPermit2(): AddLiquidityBuildCallOutput {
+        throw buildCallWithPermit2ProtocolVersionError;
+    }
```

Comment:
Yeah, I added those to remove some lint warnings.
I agree it's a bit weird, because the code should never reach those implementations.
I'd say this highlights one odd thing about the design, which is that all layers implement `AddLiquidityBase` (i.e. `AddLiquidity`, `AddLiquidityV2` and all stable/weighted/etc. variations below it).
I don't have a clear idea on how to fix/improve this 😬 
Let me know if you have any suggestions 🙃

Reply:
Ok that makes sense thanks for explaining :pray: